index/scorch/segment/zap/contentcoder.go

```diff
@@ -86,6 +86,19 @@ func (c *chunkedContentCoder) Reset() {
 	c.chunkMeta = c.chunkMeta[:0]
 }
 
+func (c *chunkedContentCoder) SetChunkSize(chunkSize uint64, maxDocNum uint64) {
+	total := int(maxDocNum/chunkSize + 1)
+	c.chunkSize = chunkSize
+	if cap(c.chunkLens) < total {
+		c.chunkLens = make([]uint64, total)
+	} else {
+		c.chunkLens = c.chunkLens[:total]
+	}
+	if cap(c.chunkMeta) < total {
+		c.chunkMeta = make([]MetaData, 0, total)
+	}
```

Comment:
blind comment here - why isn't an else part here for `c.chunkMeta = c.chunkMeta[:total]`

Reply:
The reason is that chunkMeta is used differently than chunkLens.  With chunkLens, the length of the slice must be correct, because we directly index into it, like `c.chunkLens[something] = blah`.  However, chunkMeta is always appended to, so length should be 0 initially (and that is already taken care of by Reset in the case where there is already sufficient capacity).  In the case where there is insufficient capacity, we use the 3-arg form of make to set len to 0.

It is confusing, but I believe correct.